base/broadcast.jl

```diff
@@ -284,6 +284,14 @@ eltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
 eltypestuple(a, b...) = (Base.@_pure_meta; Tuple{eltypestuple(a).types..., eltypestuple(b...).types...})
 _broadcast_eltype(f, A, Bs...) = Base._return_type(f, eltypestuple(A, Bs...))
 
+typeeltypestuple(a) = (Base.@_pure_meta; Tuple{Type{eltype(a)}})
+typeeltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{Type{T}}})
+typeeltypestuple(a, b...) = (Base.@_pure_meta; Tuple{typeeltypestuple(a).types..., typeeltypestuple(b...).types...})
+function null_safe_eltype_op(op, xs...)
+    Base.@_pure_meta
+    null_safe_op(op, typeeltypestuple(xs...))
+end
```

Comment:
I'm cautiously optimistic that simply removing `Base.@_pure_meta` is OK, and inference can infer everything anyway. So far I haven't noticed any inference regressions and `make test` still passes. Could you/someone else run nanosoldier to make sure nothing regresses?